blockmanager.go

```diff
@@ -1744,6 +1744,19 @@ func (b *blockManager) FilterHeaderTip() uint32 {
 	return b.filterHeaderTip
 }
 
+// SynchronizeFilterHeaders allows the caller to execute a function closure
+// that depends on synchronization with the current set of filter headers. This
+// allows the caller to execute an action that depends on the current filter
+// header state, thereby ensuring that the state would shift from underneath
+// them. Each execution of the closure will have the current filter header tip
+// passed in to ensue that the caller gets a consistent view.
+func (b *blockManager) SynchronizeFilterHeaders(f func(uint32) error) error {
+	b.newHeadersMtx.RLock()
```

Comment:
should be `newFilterHeadersMtx`?

Reply:
Yep! switched over and things broke, working on getting it back to a workable state...